app/helpers/promos_helper.rb

```diff
@@ -60,16 +60,11 @@ def base_referral_url
 
   def on_channel_type(channel)
     case channel.details_type
-    when "YoutubeChannelDetails"
-      "#{channel.publication_title.upcase} #{t("promo.shared.on_youtube")}"
-    when "TwitchChannelDetails"
-      "#{channel.publication_title.upcase} #{t("promo.shared.on_twitch")}"
-    when "TwitterChannelDetails"
-      "#{channel.publication_title.upcase} #{t("promo.shared.on_twitter")}"
     when "SiteChannelDetails"
       "#{channel.publication_title.upcase}"
     else
-      raise
+      integration_name = channel.details_type.gsub("ChannelDetails", "").downcase
```

Comment:
Might be able to leverage it here too https://github.com/brave-intl/publishers/blob/a5918610172eb28fea0c71e45e2438d9f4f0c87c/app/models/channel.rb#L306-L308